src/pages/docs/kubernetes/targets/kubernetes-agent/troubleshooting/sha1-certificate-incompatibility.md

```diff
@@ -0,0 +1,46 @@
+
+
+## Compatibility Issue: SHA1 on Windows Server 2012 R2
+If your Octopus Server is still using a SHA1 certificate and is running on Windows Server 2012 R2, you may encounter issues when trying to connect with the Kubernetes Agent. This is due to a known incompatibility between the agent and older Windows systems using SHA1 certificates.
+
+## Why This Happens
+Windows Server 2012 R2 lacks support for certain modern cipher suites and hash algorithms required by the Kubernetes Agent. Specifically, the agent fails during SSL negotiation because it cannot find the necessary SHA1RSA signature hash algorithm in the system's cryptographic libraries. This results in a failure to establish a secure connection between the Tentacle Agent and the Octopus Server.
+
+## Diagnosis
+To confirm that the issue is caused by SHA1 compatibility, check each of the following signs:
+1. **Verify the OS**: Ensure the Octopus Server is running on Windows Server 2012 R2.
+2. **Check the Certificate Type**: Confirm that the Octopus Server is using a SHA1 certificate. Verify that the Octopus Server is using a SHA1 certificate. The simplest method is to query the Octopus REST API endpoint `/api/certificates/certificate-global` and inspect the **SignatureAlgorithm** field.
+3. **Enable Schannel Event Logging**: On the Windows Server, [enable Schannel event logging](https://learn.microsoft.com/en-us/troubleshoot/developer/webapps/iis/health-diagnostic-performance/enable-schannel-event-logging). Look for an error event stating:
+`An TLS 1.2 connection request was received from a remote client application, but none of the cipher suites supported by the client application are supported by the server. The SSL connection request has failed.`
+This error will occur when the agent attempts to connect to the server, typically during agent startup or installation.
+
+:::figure
+![Windows Event Viewer Schannel Error](/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/kubernetes-agent-troubleshooting-sha1-windows-event-viewer.png)
+:::
+
+## Solutions
+To resolve this issue and continue using the Kubernetes Agent with your Octopus Server, you have two options, both of which are recommended regardless of the current issue:
+
+### Option 1: Rotate the SSL certificate
+Replace the current SHA1 certificate with a more secure **SHA256** certificate. This will ensure that the Tentacle Agent can successfully complete the SSL handshake with the server.
+
+#### Steps to rotate the certificate
+1. Generate a new self-signed certificate using the SHA256RSA algorithm.
+2. Install the new certificate on your Octopus Server.
+3. Update all the associated Tentacles to trust the new certificate.
```

Comment:
There is but it isn't documented. You'd have to fiddle with the thumbprint that is stored in the agent's config map. 

However, here my assumption is that only the non-Kubernetes Tentacles would need to be updated to trust the new certificate, because the agent wouldn't have worked with the sha1 server certificate to begin with.

Reply:
Fair 👍